### PR TITLE
Add `ci_method` for CIs in `report_sample()`

### PR DESCRIPTION
**Pull request by strengejacke**

### CODE REVIEW

**Comment by strengejacke** [comment]
@easystats/core-team Can you check whether the formulas are correctly implemented?

I'm using eq. 1 for "wald" and eq. 3 for "wilson":

image

`p_hat` is p:

```r
  if (ci_method == "wilson") {
    prop <- (2 * n * p_hat) + z^2
    se <- z * sqrt((z^2 + 4 * n * p_hat * (1 - p_hat)))
    out <- c(prop - se, prop + se) / (2 * (n + z^2))
  } else {
    se <- z * suppressWarnings(sqrt(p_hat * (1 - p_hat) / n))
    out <- c(p_hat- se, p_hat+ se)
  }
```

**Comment by strengejacke** [comment]
from the R help `?report_sample`:

image

File: R/report_sample.R [hunk 1]
```diff
@@ -315,7 +353,17 @@ report_sample <- function(data,
...
-  stats::qnorm((1 + ci) / 2) * suppressWarnings(sqrt(p_hat * (1 - p_hat) / n))
+  z <- stats::qnorm((1 + ci) / 2)
+
+  if (ci_method == "wilson") {
+    prop <- (2 * n * p_hat) + z^2
+    se <- z * sqrt(z^2 + 4 * n * p_hat * (1 - p_hat))
+    out <- c(prop - se, prop + se) / (2 * (n + z^2))
+  } else {
+    se <- z * suppressWarnings(sqrt(p_hat * (1 - p_hat) / n))
+    out <- c(p_hat - se, p_hat + se)
```
Comment (bwiernik) [inline R/report_sample.R:364]:
```suggestion
    out <- c(p_hat - moe, p_hat + moe)
```

**Comment by strengejacke** [comment]
image

**Comment by strengejacke** [comment]
``` r
library(report)
set.seed(123)
d <- data.frame(x = factor(sample(1:3, 100, TRUE, c(.01, .39, .6))))

# ups, negative CI
report::report_sample(d, ci = 0.95, ci_method = "wald")
#> # Descriptive Statistics
#> 
#> Variable |           Summary
#> ----------------------------
#> x [1], % |   1.0 (-1.0, 3.0)
#> x [2], % | 39.0 (29.4, 48.6)
#> x [3], % | 60.0 (50.4, 69.6)
report::report_sample(d, ci = 0.95, ci_method = "wald", ci_correct = TRUE)
#> # Descriptive Statistics
#> 
#> Variable |           Summary
#> ----------------------------
#> x [1], % |   1.0 (-1.5, 3.5)
#> x [2], % | 39.0 (28.9, 49.1)
#> x [3], % | 60.0 (49.9, 70.1)

report::report_sample(d, ci = 0.95, ci_method = "wilson")
#> # Descriptive Statistics
#> 
#> Variable |           Summary
#> ----------------------------
#> x [1], % |    1.0 (0.2, 5.4)
#> x [2], % | 39.0 (30.0, 48.8)
#> x [3], % | 60.0 (50.2, 69.1)
report::report_sample(d, ci = 0.95, ci_method = "wilson", ci_correct = TRUE)
#> # Descriptive Statistics
#> 
#> Variable |           Summary
#> ----------------------------
#> x [1], % |    1.0 (0.1, 6.2)
#> x [2], % | 39.0 (29.6, 49.3)
#> x [3], % | 60.0 (49.7, 69.5)
```

<sup>Created on 2023-03-31 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by bwiernik** [comment]
Nice!

APA style for intervals is square brackets rather than round brackets if we want to follow that

**Comment by strengejacke** [comment]
Yeah, we can do that

**Comment by strengejacke** [comment]
But APA...

https://user-images.githubusercontent.com/26301769/229223922-66797988-df56-482c-8c53-10a7f71cb9a8.mp4

I mean, psychologists are those with this weird ANOVA stuff, or not?

**Pull request merged**